sopel/modules/admin.py

```diff
@@ -239,8 +285,16 @@ def hold_ground(bot, trigger):
     WARNING: This may not be needed and could cause problems if Sopel becomes
     annoying. Please use this with caution.
     """
-    if bot.config.admin.hold_ground and bot.nick == trigger.args[1]:
+    channel = trigger.sender
+    if bot.nick != trigger.args[1]:
+        # not the bot; ignore
+        return
+
+    if bot.config.admin.hold_ground:
```

Comment:
Hi, I'm micro-optimizing memory allocations in an interpreted language again.

```suggestion
    if bot.nick != trigger.args[1]:
        # not the bot; ignore
        return

    channel = trigger.sender
    if bot.config.admin.hold_ground:
```